sdk/iot/azure-iot-modelsrepository/azure/iot/modelsrepository/_client.py

```diff
@@ -0,0 +1,236 @@
+    _resolver,
+    _pseudo_parser,
+    _constants,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# Public constants exposed to consumers
+DEPENDENCY_MODE_TRY_FROM_EXPANDED = "tryFromExpanded"
+DEPENDENCY_MODE_DISABLED = "disabled"
+DEPENDENCY_MODE_ENABLED = "enabled"
+
+
+# Convention-private constants
+_DEFAULT_LOCATION = "https://devicemodels.azure.com"
+_REMOTE_PROTOCOLS = ["http", "https"]
+_TRACE_NAMESPACE = "modelsrepository"
+
+
+class ModelsRepositoryClient(object):
+    """Client providing APIs for Models Repository operations"""
+
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        """
+        :keyword str repository_location: Location of the Models Repository you wish to access.
+            This location can be a remote HTTP/HTTPS URL, or a local filesystem path.
+            If omitted, will default to using "https://devicemodels.azure.com".
+        :keyword str dependency_resolution: Dependency resolution mode.
+            Possible values:
+                - "disabled": Do not resolve model dependencies
+                - "enabled": Resolve model dependencies from the repository
+                - "tryFromExpanded": Attempt to resolve model and dependencies from an expanded
+                        DTDL document in the repository. If this is not successful, will fall back
+                        on manually resolving dependencies in the repository
+            If using the default repository location, the default dependency resolution mode will
+            be "tryFromExpanded". If using a custom repository location, the default dependency
+            resolution mode will be "enabled".
+        :keyword str api_version: The API version for the Models Repository Service you wish to
+            access.
+
+        For additional request configuration options, please see [core options](https://aka.ms/azsdk/python/options).
+
+        :raises: ValueError if an invalid argument is provided
+        """
+        repository_location = kwargs.get("repository_location", _DEFAULT_LOCATION)
+        _LOGGER.debug("Client configured for respository location %s", repository_location)
+
+        self.resolution_mode = kwargs.get(
+            "dependency_resolution",
+            DEPENDENCY_MODE_TRY_FROM_EXPANDED
+            if repository_location == _DEFAULT_LOCATION
+            else DEPENDENCY_MODE_ENABLED,
+        )
+        if self.resolution_mode not in [
+            DEPENDENCY_MODE_ENABLED,
+            DEPENDENCY_MODE_DISABLED,
+            DEPENDENCY_MODE_TRY_FROM_EXPANDED,
+        ]:
+            raise ValueError("Invalid dependency resolution mode: {}".format(self.resolution_mode))
+        _LOGGER.debug("Client configured for dependency mode %s", self.resolution_mode)
+
+        # NOTE: depending on how this class develops over time, may need to adjust relationship
+        # between some of these objects
+        self.fetcher = _create_fetcher(location=repository_location, **kwargs)
```

Comment:
Should the instance bound fetcher and resolver also have an underscore prefix/private convention like `_psuedo_parser`?

Reply:
I only made attributes convention-prviate for the client itself, for the inner classes I did not add convention-private attributes as they are not part of the customer-facing surface, and the SDK guidelines do not require me to do so, nor is it particularly common to rigidly enforce this convention on attributes.

I did choose to use convention-private methods on inner classes, because I find those helpful during development. 

If you feel particularly strongly on this point, I can go make everything convention private.

Reply:
I don't. We can consider this resolved.